vault/sources/Vault.move

```diff
@@ -168,37 +192,36 @@ module titusvaults::Vault {
     }
 
     /// to create new vaults for Nth round
-    public fun create_vault<VaultT, AssetT>(_host: &signer) acquires RoundState {
+    public fun create_vault<VaultT, AssetT>(_host: &signer) acquires VaultMap {
         let host_addr = address_of(_host);
         assert!(host_addr == @titusvaults, E_NOT_AUTHORIZED);
+        
+        let vault_map = borrow_global_mut<VaultMap>(@titusvaults);
+        let current_round = vector::borrow_mut(&mut vault_map.rounds, 0);  
 
-        let current_round = borrow_global<RoundState>(@titusvaults);
-        if (!exists<VaultMap>(host_addr)) {
-            move_to(_host, VaultMap {
-                deposits: smart_table::new(),
-                rounds: vector::empty<RoundState>(),
-                total_rounds: 0
-            });
-            move_to(_host, Vault<VaultT, AssetT> {
-                coin_store: coin::zero(),
-                creation_time: timestamp::now_microseconds(),
-                creation_round: current_round.current_round_id,
-                total_shares: 0
-            });
-        };
+        move_to(_host, Vault<VaultT, AssetT> {
+            coin_store: coin::zero(),
+            creation_time: timestamp::now_microseconds(),
+            creation_round: current_round.current_round_id,
+            total_shares: 0
+        });
     }
 
     // deposit vault
-    public (friend) fun deposit_vault<VaultT, AssetT>( account: &signer, _coin: Coin<AssetT>) acquires Vault, VaultMap, RoundState {
+    public (friend) fun deposit_vault<VaultT, AssetT>( account: &signer, _coin: Coin<AssetT>) acquires Vault, VaultMap {
         let user_addr = address_of(account);
         let vault = borrow_global_mut<Vault<VaultT, AssetT>>(@titusvaults);
         let vault_map = borrow_global_mut<VaultMap>(@titusvaults);
-        let round_state = borrow_global_mut<RoundState>(@titusvaults);
+
+        // check if there is at least one round in the vault_map.rounds vector
+        assert!(vector::length(&vault_map.rounds) >= 1, E_INVALID_OPERATION);
+
+        let first_round_state = vector::borrow_mut(&mut vault_map.rounds, 0); 
```

Comment:
we need to perform deposit on round ID whose ID has been passed to this function as parameter and not the first round ID.